zerver/migrations/0182_set_initial_value_is_private_flag.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+import sys
+from django.db import migrations
+
+from django.db import migrations, transaction
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import F
+
+def set_initial_value_of_is_private_flag(
+        apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    UserMessage = apps.get_model("zerver", "UserMessage")
+    Message = apps.get_model("zerver", "Message")
+    i = 0
+    total = Message.objects.count()
```

Comment:
This `total` approach is buggy, because if messages are being sent while this runs, the total we queried before is wrong.   At least if we're using `atomic=False` which we should for this expensive migration.

Reply:
(Using the looping flow from migration 0177 is robust against this sort of thing)

Reply:
(we need to adapt it a bit for the fact the query is sparse, though; see the code I just pushed)